src/proxy-ui-api/src/main/java/org/niis/xroad/securityserver/restapi/openapi/ClientsApiController.java

```diff
@@ -575,14 +583,16 @@ private ClientType getClientType(String encodedId) {
     @PreAuthorize("hasAuthority('VIEW_CLIENT_ACL_SUBJECTS')")
     public ResponseEntity<Set<ServiceClient>> getClientServiceClients(String id) {
         ClientId clientId = clientConverter.convertId(id);
-        Set<ServiceClient> serviceClients = null;
+        SortedSet<ServiceClient> sortedServiceClients = null;
         try {
-            serviceClients = serviceClientConverter.
+            Set<ServiceClient> serviceClients = serviceClientConverter.
                     convertServiceClientDtos(serviceClientService.getServiceClientsByClient(clientId));
+            sortedServiceClients = new TreeSet<>(serviceClientSortingComparator);
+            sortedServiceClients.addAll(serviceClients);
```

Comment:
The converter creates a HashSet that is immediately converted to a TreeSet. Both data structures have some overhead, so it would be more efficient to make the converter directly return an ordered set (e.g. by providing the result set as a parameter, or ordering the input list and returning a LinkedHashSet).

Reply:
The implementation was moved to the Converter that orders the input list and returns a LinkedHashSet.